src/game/player/skin/basic.rs

```diff
@@ -179,10 +179,15 @@ impl PlayerSkin for BasicPlayerSkin {
         }
     }
 
-    fn set_state(&mut self, state: PlayerAnimationState) {
+    fn set_state(&mut self, state: PlayerAnimationState, curr_tick: u16) {
         if self.state != state {
             self.state = state;
-            self.tick = 0;
+
+            //self.tick = 0; // this should not happen
+            //self.tick = curr_tick; // this should happen instead, but there's a problem with ticking on 4 that results in an instant 1st frame animation.
+            // this dirty hack should fix that.
+            self.tick = if curr_tick % 5 == 4 {u16::MAX} else {curr_tick};
```

Comment:
>  so your code will always set `self.tick` to `u16::MAX`

What? My code will only set `self.tick` to `u16::MAX` if `anim_counter` is 4, which is a state that happens 1 in 5 times.
I'll admit, I wrote this some time ago, the `%5` may actually be redundant, but it is there for number safety.

---
The changes I proposed here may not be the most clear, so let me explain what's happening:

`self.anim_counter` will range between 0 and 4, and will never go above 4.

`skin.tick` does not have an upper bound. It will keep increasing so long as the player isn't changing states.

The problem is that when `skin.set_state` is called in the original code, `skin.tick` is always reset to 0, but `self.anim_counter` is not. The player's gun and step sounds are tied to `self.anim_counter`, but the player's rect is set to `skin.tick`. This results in a desync between SFX and sprite state.

To solve this problem, I passed `self.anim_counter` to `skin.tick` so that they would have the same delay and be synched with each other.

When we set the player skin's `tick` variable, there is a small chance we do when `self.anim_counter` is 4. If we **didn't** set `skin.tick` to roll over to 0 next time it is encremented, `skin.tick` **would** be set to 5, which is troublesome in light of the way the player skin indexes are accessed, which is with:
`(tick as usize / 5) % 4`, which will yield an index of `1` instead of `0`, which is not what we want.

This means that while the sound and gun offset would start on frame 0 with a full wait, the player rect would instantly start at frame 1 with a full wait.

*Once again, you can best see the changes in action if you try out a copy of the modified code. (This particular bug can much more easily be seen if tick rate is slowed down to 0x1x).*